homeassistant/components/flo/device.py

```diff
@@ -0,0 +1,176 @@
+    def available(self) -> bool:
+        """Return True if device is available."""
+        return self._available
+
+    @property
+    def current_system_mode(self) -> str:
+        """Return the current system mode."""
+        return self._current_system_mode
+
+    @property
+    def target_system_mode(self) -> str:
+        """Return the target system mode."""
+        return self._target_system_mode
+
+    @property
+    def current_flow_rate(self) -> float:
+        """Return current flow rate in gpm."""
+        return self._current_flow_rate
+
+    @property
+    def current_psi(self) -> float:
+        """Return the current pressure in psi."""
+        return self._current_psi
+
+    @property
+    def temperature(self) -> float:
+        """Return the current temperature in degrees F."""
+        return self._temperature
+
+    @property
+    def consumption_today(self) -> float:
+        """Return the current consumption for today in gallons."""
+        return self._consumption_today
+
+    @property
+    def firmware_version(self) -> str:
+        """Return the firmware version for the device."""
+        return self._firmware_version
+
+    async def _update_device(self, *_) -> None:
+        """Update the device information from the API."""
+        device_information = await self.api_client.device.get_info(self._flo_device_id)
+        _LOGGER.debug("Flo device data: %s", device_information)
+        self._mac_address: str = device_information["macAddress"]
+        self._model: str = device_information["deviceModel"]
+        self._device_type: str = device_information["deviceType"]
+        self._serial_number: str = device_information["serialNumber"]
+        self._available = device_information["isConnected"]
+        self._firmware_version = device_information["fwVersion"]
+        self._rssi = device_information["connectivity"]["rssi"]
+        self._last_heard_from_time = device_information["lastHeardFromTime"]
+        self._current_system_mode = device_information["systemMode"]["lastKnown"]
+        self._target_system_mode = device_information["systemMode"]["target"]
+        self._current_flow_rate = device_information["telemetry"]["current"]["gpm"]
+        self._current_psi = device_information["telemetry"]["current"]["psi"]
+        self._temperature = device_information["telemetry"]["current"]["tempF"]
+
+    async def _update_consumption_data(self, *_) -> None:
+        """Update water consumption data from the API."""
+        today = dt_util.now().date()
+        start_date = datetime(today.year, today.month, today.day, 0, 0)
+        end_date = datetime(today.year, today.month, today.day, 23, 59, 59, 999000)
+        water_usage = await self.api_client.water.get_consumption_info(
+            self._flo_location_id, start_date, end_date
+        )
+        _LOGGER.debug("Updated Flo consumption data: %s", water_usage)
+        self._consumption_today = water_usage["aggregations"]["sumTotalGallonsConsumed"]
```

Comment:
Any reason we shouldn't just store the whole object too?

Reply:
updated